obsreport/obsreporttest/obsreporttest.go

```diff
@@ -41,17 +45,35 @@ var (
 	processorTag, _ = tag.NewKey("processor")
 )
 
+type TestTelemetrySettings struct {
+	component.TelemetrySettings
+	SpanRecorder           *tracetest.SpanRecorder
+	ExporterCreateSettings component.ExporterCreateSettings
```

Comment:
I think we should have multiple funcs to create all types of create settings.

```
func (tts *TestTelemetrySettings) ToExporterCreateSettings() ExporterCreateSettings {
}

func (tts *TestTelemetrySettings) ToReceiverCreateSettings() ReceiverCreateSettings {
}
```

Reply:
I think we should not ask for any argument like BuildInfo in case we will add more so treat all the same, user can set other settings after the call.